sorting/merge_sort/merge_sort.py

```diff
@@ -0,0 +1,38 @@
+def merge(left, right):
+    """
+    Merge two sorted array into one
+    Parameters
+    ----------
+        left:   list
```

Comment:
one option is to use Python 3's type hints:
```python
from typing import List
...
def merge(left: List, right: List):
    ...
```
It's clear and fully supported by modern IDE's (which helps a lot to write reliable code)
Obviously, you still have to describe non-trivial arguments in docstring, though